src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -143,7 +143,7 @@ public string GenerateStartTag()
         /// </returns>
         public string GeneratePreContent()
         {
-            if (SelfClosing)
+            if (!string.IsNullOrWhiteSpace(TagName) && SelfClosing)
```

Comment:
- put another way, revert changes to this file
- it may be worth making a different change: update the doc comments for the `SelfClosing` property to include `<remarks/>` that mention `PreContent`, `Content`, and `PostContent` are ignored when `SelfClosing` is `true`.